lib/JSON/Validator.pm

```diff
@@ -420,31 +420,41 @@ sub _resolve {
   return $schema;
 }
 
+sub _location_to_abs {
+  my ($location, $base) = @_;
+  my $location_as_url = Mojo::URL->new($location);
+  return $location_as_url if $location_as_url->is_abs;
+  # definitely relative now
+  if ($base->isa('Mojo::File')) {
+    return $base if !length $location;
+    return $base->sibling(split '/', $location)->realpath;
+  }
+  return $location_as_url->to_abs($base);
```

Comment:
This is a minor thing that I can do myself, but I guess you want credit for your own lines of code. How about something like this:

```
sub _location_to_abs {
  my ($location, $base) = @_;
  my $location_as_url = Mojo::URL->new($location);
  return $location_as_url if $location_as_url->is_abs;    # Already absolute
  return $location_as_url->to_abs($base) unless $base->isa('Mojo::File');    # Mojo::URL to absolute
  return $base unless length $location;                                      # No sibling file to locate
  return $base->sibling(split '/', $location)->realpath;                     # Find sibling file relative to $base
}
```

Reply:
(not sure if the comments are needed)

Reply:
I think your version is better and am pleased if you update it to that. I think the comments do add something.